### ISSUE TITLE
Use qualified function names

### ISSUE THREAD

**Issue by simonwahlgren** [DataDog/dd-trace-py#1943]
Hi!

At the moment it's a bit hard to understand which resource is being traced when you have multiple methods with the same name in the same file. For example:

```python
class FooView:
    @tracer.wrap()
    def get(self):
        pass


class BarView:
    @tracer.wrap()
    def get(self):
        pass
```

The above code would currently use the same default span name `get` for both functions.

I'm proposing just using the function qualified name instead but I'm not sure if there are any drawbacks of doing this? 

I have noticed local functions get "weird" function names, for example:
```python
def app(tracer):
    app = Sanic(__name__)

    @tracer.wrap()
    async def random_sleep():
        await asyncio.sleep(random.random())
```
`random_sleep` would get a qualified name of `app.<locals>.random_sleep` not sure if this is a problem or not since it's always possible to override it.

What do you guys think?

**EDIT**: Seems like qualified names are not supported in Python 2 (introduced in 3.3) so have to add support for that somehow.

**Comment by Kyle-Verhoog** [comment]
@simonwahlgren yeah I see the issue. Having the fully qualified name makes sense to me. I think the local case is acceptable although it'd result in the same name clashing if there were multiple functions defined in local scope in the same module. I don't think there's a nice way around this however.

That brings the idea to mind of attaching the file name + line number of the function as a tag on the span 🤔. Might be useful in addition to or in place of fully qualified names. What do you think @simonwahlgren?

**Comment by simonwahlgren** [comment]
@Kyle-Verhoog I agree, I think the case with multiple functions with the same name in the same scope is a rare edge case, the only way to differentiate them I guess would be to include the line number as you say. Also most linters will most probably return an error.

I also like the idea of adding the file name + line number as a span tag. Where would be the best place to add those tags?

Also do you have any thoughts regarding being Python 2 compliant? Use the current implementation as a fallback might be good enough maybe?

**Comment by Kyle-Verhoog** [comment]
@simonwahlgren the file name + line number can be addressed in a separate PR from #1942 and might require some more thought. 

As for Python 2 it'd be nice to be consistent between the two but not essential. Falling back/keeping the existing functionality would be fine 🙂
